### ISSUE TITLE
Can't set date when using setSelectableDays(Calendar[] )

### ISSUE THREAD

**Issue by anaszakariyah** [wdullaer/MaterialDateTimePicker#243]
Here is part of my code.
```
Calendar[] selectableDate = ... //contains now calendar

Calendar now = Calendar.getInstance();
 DatePickerDialog datePickerDialog = DatePickerDialog.newInstance(this,
                now.get(Calendar.YEAR),
                now.get(Calendar.MONTH),
                now.get(Calendar.DAY_OF_MONTH));
 datePickerDialog.setSelectableDays(selectableDate);
 datePickerDialog.show(getFragmentManager(), "DatePickerDialog");`
```
There is current date in selectableDate array. In this case I want to select current date. How can I set the dialog to show desired date? With this code date selected will always the "first" selectable date. I try to use the datePickerDialog.initialize(...currentDate) after setSelectableDays() the result is still the same.

**Comment by wdullaer** [comment]
Your initial date needs to be selectable.
If it's not, the library will round it to the nearest selectable one. This check is there to ensure that you can't initialize the dialog in an inconsistent state.

**Comment by wdullaer** [comment]
You'll need to save it in the callback and then use that value when you construct the new Dialog.
These pickers are not meant to be reused, you should create a new instance every time you need one.

**Comment by anaszakariyah** [comment]
I am creating new Instance every time I make the dialog. It works fine without selectabledays and it shows the saved date, But using the selectableDays wont do. It just select the very first selectableday available.

**Comment by wdullaer** [comment]
Are you sure the currently saved day is selectable?

**Comment by anaszakariyah** [comment]
Hi. I finally solved this issue. Sorry for long reply because I've been working on project (including this library too *big thanks to you wdullaer :) ) almost non stop and got no time to check github.

I dont know how the rounding of selectableDays works. But I've got no issues if I use the exact date (minimal in days, without the hours, minutes or so). Before I got the initial calendar from new Date(), so obviously it will have the current hours, minutes, etc and thats how the date wont have the exact match. So I use Simple Date format for this issue in example like dd/mm/yyyy then I get the calendar from that and voila the selectableDays work and it can save the state too.

Thank you for this wonderful library wdullaer :)
